site/app/libraries/database/DatabaseQueries.php

```diff
@@ -773,7 +773,10 @@ public function splitPost($post_id, $title, $categories_ids) {
         return [$old_thread_id, $thread_id, $counted_posts];
     }
 
-    public function createPost($user, $content, $thread_id, $anonymous, $type, $first, $hasAttachment, $markdown, $parent_post = -1) {
+    /**
+     * @param string $attachment_name
+     */
+    public function createPost($user, $content, $thread_id, $anonymous, $type, $first, $hasAttachment, $markdown, $attachment_name, $parent_post = -1) {
```

Comment:
Please add native PHP type annotations to (at least) the new parameters.  Apply throughout.